src/core_memusage.h

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2015 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_CORE_MEMUSAGE_H
+#define BITCOIN_CORE_MEMUSAGE_H
+
+#include "primitives/transaction.h"
+#include "primitives/block.h"
+#include "memusage.h"
+
+static inline size_t RecursiveDynamicUsage(const CScript& script) {
+    return memusage::DynamicUsage(*static_cast<const CScriptBase*>(&script));
```

Comment:
I'm not sure we need static_cast here, CScriptBase is a parent class for CScript, so it can be assigned without casting. The same about non-const to const.

Reply:
I wouldve said the same, and even attempted it in a different part of the code. But from what Ive found most functions that take CScriptBase don't accept CScript without the cast.

Reply:
Again, if it's a merge, then we can leave it like that, I'm curious why doesn't it work, but they probably have already tested this code :)